.buildkite/commands/check-script-permissions.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash -eu
+
+for file in bin/*; do
+  echo "Checking for +x permission on $file..."
+  if ! [ -x "$file" ]; then
+    echo "$file is not executable!"
+    exit 1
+  fi
+done
```

Comment:
Actually, now that I think about it, this would be a good fit to put into the `bats` tests, wdyt?

Like
```sh
@test "Ensure all commands have +x permission" {
  for file in bin/*; do
    [ -x "$file"]
  done
}
```

Reply:
Sure, we might as well just add another test instead of the build step like I initially did.

Reply:
I've replaced the build step with the bats test in 37288d74d80cc2bdf9f0913f02c6de3e6c61aacc